ed_sensor_integration/src/kinect/image_buffer.cpp

```diff
@@ -119,38 +121,15 @@ bool ImageBuffer::nextImage(const std::string& root_frame, rgbd::ImageConstPtr&
         image_buffer_.pop();
```

Comment:
I think here is where you want to drop images that are too old.

The question is also what should be the behaviour? Should the image buffer keep iterating till it finds a valid image and then return that one. Or should the user accept the false return and keeps calling `nextImage` till it gets what is wants.

Reply:
upon a call to NextImage a client wants the most recent available combination of image and pose, or None if the most recent one is identical to the previous call of the NextImage function. 
The weird behaviour previously came from two things:
1) the assumption that the function is called regularly, which was used to keep the buffer empty
2) Apparently there is no criteria on how old this 'most recent' result may be. Which can defy expectations